nova_vm/src/ecmascript/execution/agent.rs

```diff
@@ -66,7 +91,7 @@ pub struct Agent {
 }
 
 impl Agent {
-    pub fn new(options: Options, host_hooks: &'static dyn HostHooks) -> Self {
+    pub(crate) fn new(options: Options, host_hooks: &'static dyn HostHooks) -> Self {
```

Comment:
**suggestion:** Currently `Options` has a `disable_gc` field, which AFAICT is completely ignored. If we want to provide the option of not allowing GC for an agent, we could instead make `Agent::new` public, where the choice between `BoxedAgent` and `Agent` would determine whether GC is allowed or not.

Reply:
I made `gc()` a no-op if `disable_gc` is turned on.

Returning a different type depending on a parameter would require a return type enum, or a return-position Trait impl.

Reply:
What I meant is that, since the only difference in when you'd use `GCAgent` vs `Agent` is if you're going to use GC, we could remove the `disable_gc` option, since that is already encoded in the choice of type.